homeassistant/components/ambient_station/sensor.py

```diff
@@ -0,0 +1,115 @@
+"""
+Support for Ambient Weather Station Service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.ambient_station/
+"""
+import logging
+
+from homeassistant.components.ambient_station import SENSOR_TYPES
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import ATTR_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import (
+    ATTR_LAST_DATA, DATA_CLIENT, DOMAIN, TOPIC_UPDATE, UNITS_SI, UNITS_US)
+
+DEPENDENCIES = ['ambient_station']
+_LOGGER = logging.getLogger(__name__)
+
+UNIT_SYSTEM = {UNITS_US: 0, UNITS_SI: 1}
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up an Ambient PWS sensor based on existing config."""
+    pass
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up an Ambient PWS sensor based on a config entry."""
+    ambient = hass.data[DOMAIN][DATA_CLIENT][entry.entry_id]
+
+    if ambient.unit_system:
+        sys_units = ambient.unit_system
+    elif hass.config.units.is_metric:
+        sys_units = UNITS_SI
+    else:
+        sys_units = UNITS_US
+
+    sensor_list = []
+    for mac_address, station in ambient.stations.items():
+        for condition in ambient.monitored_conditions:
+            name, unit = SENSOR_TYPES[condition]
+            if isinstance(unit, list):
+                unit = unit[UNIT_SYSTEM[sys_units]]
+
+            sensor_list.append(
+                AmbientWeatherSensor(
+                    ambient, mac_address, station[ATTR_NAME], condition, name,
+                    unit))
+
+    async_add_entities(sensor_list, True)
+
+
+class AmbientWeatherSensor(Entity):
```

Comment:
Should this be a sensor or weather platform?

Reply:
Would love your opinion:

1. The original `ambient_station` platform was a sensor platform.
2. There are several areas where the overlap is missing (i.e., Ambient _doesn't_ return any sort of forecast, _does_ return a lot of properties not in the weather component, etc).

That said, I'm open!

Reply:
If it's a bad overlap with weather, I'm ok with sensor.